In [10]:
### Problem Set 1
### Jolie L'Heureux
### jel2227
### Caiti Levin helped me with this PSET
### Alex Savino helped me with this PSET
### Leah Mcgee-Gold helped with problem 2
### My CS friend Thomas Bueler also helped me with an error in problem 2 <3
import numpy as np
import math as m

In [17]:
###Problem 1
x=1
while x < 10.5:
  y=m.log(x)
  print(x, "        ", y)
  x=x+0.5

1          0.0
1.5          0.4054651081081644
2.0          0.6931471805599453
2.5          0.9162907318741551
3.0          1.0986122886681098
3.5          1.252762968495368
4.0          1.3862943611198906
4.5          1.5040773967762742
5.0          1.6094379124341003
5.5          1.7047480922384253
6.0          1.791759469228055
6.5          1.8718021769015913
7.0          1.9459101490553132
7.5          2.0149030205422647
8.0          2.0794415416798357
8.5          2.1400661634962708
9.0          2.1972245773362196
9.5          2.2512917986064953
10.0          2.302585092994046


In [33]:
###Problem 2

import numpy as np
import math as m


def flux(p,z):
    
    if z > (1+p):
        lambdaa = 0
    elif abs(1-p) < z <= 1+p:
        k_1 = m.acos((1-p**2+z**2)/(2*z)) 
        k_0 = m.acos((p**2+z**2-1)/(2*p*z))

        eq_sqrt = m.sqrt(abs((4*z**2-(1+z**2-p**2)**2)/(4)))
        x_eq=(1/m.pi)*(p**2*k_0+k_1-eq_sqrt)  
        
        lambdaa = x_eq
    elif z <= 1-p:
        lambdaa=p**2
    elif z <= p-1:
        lambdaa=1
    return (1-lambdaa)




for z_values in np.arange(-1.2,1.2,0.05):
    print(flux(0.2,z_values))

0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.96
0.9599999995348759
0.9631632627651051
0.9684170049114549
0.9745022090966327
0.9808496779305926
0.9870426732421561
0.9926934503699997
0.9973352639830162


In [34]:
###Problem 3
import numpy as np
import math as m

x_N= (m.pi)/2
x_0=0
N =10
N_i= N+1

z = np.arange(0,N_i,1)

def int_rectangles (x_N, x_0, N):
    delta_x= (x_N-x_0)/(N)
    area_rectangles = 0
    total_sum=0
    i=0
    while i <= N:
        for e in z:
            x_i = x_0 +e*delta_x
            area_rectangles = m.sin(x_i)*delta_x
            total_sum += area_rectangles
            i += 1
        return total_sum
area_under_curve = int_rectangles((m.pi)/2,0,N) 
error = abs(1 - area_under_curve)
print("area under curve =", area_under_curve)
print("fractional error =", error)

area under curve = 1.076482802694102
fractional error = 0.07648280269410201


In [35]:
x_N= (m.pi)/2
x_0=0
N =10**2
N_i= N+1

z = np.arange(0,N_i,1)

def int_rectangles (x_N, x_0, N):
    delta_x= (x_N-x_0)/(N)
    area_rectangles = 0
    total_sum=0
    i=0
    while i <= N:
        for e in z:
            x_i = x_0 +e*delta_x
            area_rectangles = m.sin(x_i)*delta_x
            total_sum += area_rectangles
            i += 1
        return total_sum
area_under_curve = int_rectangles((m.pi)/2,0,N) 
error = abs(1 - area_under_curve)
print("area under curve =", area_under_curve)
print("fractional error =", error)

area under curve = 1.0078334198735823
fractional error = 0.007833419873582326


In [36]:
x_N= (m.pi)/2
x_0=0
N =10**3
N_i= N+1

z = np.arange(0,N_i,1)

def int_rectangles (x_N, x_0, N):
    delta_x= (x_N-x_0)/(N)
    area_rectangles = 0
    total_sum=0
    i=0
    while i <= N:
        for e in z:
            x_i = x_0 +e*delta_x
            area_rectangles = m.sin(x_i)*delta_x
            total_sum += area_rectangles
            i += 1
        return total_sum
area_under_curve = int_rectangles((m.pi)/2,0,N) 
error = abs(1 - area_under_curve)
print("area under curve =", area_under_curve)
print("fractional error =", error)

area under curve = 1.0007851925466311
fractional error = 0.0007851925466311194


In [37]:
x_N= (m.pi)/2
x_0=0
N =10**4
N_i= N+1

z = np.arange(0,N_i,1)

def int_rectangles (x_N, x_0, N):
    delta_x= (x_N-x_0)/(N)
    area_rectangles = 0
    total_sum=0
    i=0
    while i <= N:
        for e in z:
            x_i = x_0 +e*delta_x
            area_rectangles = m.sin(x_i)*delta_x
            total_sum += area_rectangles
            i += 1
        return total_sum
area_under_curve = int_rectangles((m.pi)/2,0,N) 
error = abs(1 - area_under_curve)
print("area under curve =", area_under_curve)
print("fractional error =", error)

area under curve = 1.0000785377601753
fractional error = 7.853776017530123e-05


In [38]:
x_N= (m.pi)/2
x_0=0
N =10**5
N_i= N+1

z = np.arange(0,N_i,1)

def int_rectangles (x_N, x_0, N):
    delta_x= (x_N-x_0)/(N)
    area_rectangles = 0
    total_sum=0
    i=0
    while i <= N:
        for e in z:
            x_i = x_0 +e*delta_x
            area_rectangles = m.sin(x_i)*delta_x
            total_sum += area_rectangles
            i += 1
        return total_sum
area_under_curve = int_rectangles((m.pi)/2,0,N) 
error = abs(1 - area_under_curve)
print("area under curve =", area_under_curve)
print("fractional error =", error)

area under curve = 1.000007853961068
fractional error = 7.853961067949555e-06


N_10 error = 0.07648280269410201
N_100 error = 0.007833419873582326
N_1000 error = 0.0007851925466311194
N_10000 error = .00007853776017530123
N_100000 error = .000007853961067949555

The fractional error decreases exponentialy as the number of rectangles increasse which makes the shceme exponential order
This can be seen by the error decreasing by a smaller amount each time

true value of the function is 1. 